Adding beagleplay support

delhiryder opened the pull request:
Initial PR - 
Got the 'balena-image' target to run to completion. Made a pair of new beaglebone devices: `beaglebone-play` and `beaglebone-play-k3r5`.
The code compiles, but does not get past the u-boot stage.
This PR was created to allow the balena dev to inspect it.

resin-jenkins commented:
Can one of the admins verify this patch?

perezmeyer commented:
Currently u-boot-ti-staging is having issues reading the EEPROM (assuming it does the same as the beaglebone-ai64). I have just pushed another MR in which I add some patches on top of it to force the board, you could do the same for the time being.It is worth to mention that the bootloader is currently work in progress in meta-ti.

perezmeyer commented on layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging_%.bbappend:
```diff
@@ -11,14 +11,10 @@ SRC_URI:append = " \
 
-SSTATE_ALLOW_OVERLAP_FILES += "${DEPLOYDIR}/extra_uEnv.txt"
-SSTATE_ALLOW_OVERLAP_FILES += "${DEPLOYDIR}/uEnv.txt_internal"
+SSTATE_ALLOW_OVERLAP_FILES += "${DEPLOY_DIR_IMAGE}/extra_uEnv.txt"
```
This is interesting, I ended up removing one of them... the question is: how do you know if it is OK to have them overlapped? What if one of them has different content? (on the bbai64 case the files where empty on both cases).

delhiryder replied:
I needed to add this because my build kept failing. The beagleplay has a 'BBMULTICONFIG +=' that makes behave a bit strangely. I am sure this can be fixed in a nicer way.

perezmeyer replied:
It probably features two architectures like the AI64

perezmeyer commented on layers/meta-balena-beaglebone/recipes-core/images/balena-image.inc:
```diff
@@ -15,13 +15,13 @@ BALENA_BOOT_PARTITION_FILES:beagleboard-xm = " \
-BALENA_BOOT_PARTITION_FILES_beaglebone-play = " \
-    MLO: \
+BALENA_BOOT_PARTITION_FILES:beaglebone-play = " \
+    ${SPL_BINARY}: \
```
I'll also take this one.

perezmeyer commented on layers/meta-balena-beaglebone/recipes-bsp/u-boot/u-boot-ti-staging_%.bbappend:
```diff
@@ -11,14 +11,10 @@ SRC_URI:append = " \
 
 
-SRC_URI:append:beaglebone-pocket = " \
-    file://0001-u-boot-Disable-logging-on-UART0.patch \
```
This probably will need to get back in some way.

delhiryder replied:
Why do we need this patch ? (I can add it back, since I now have the sources in my devtool workspace)

perezmeyer replied:
If I understand correctly Balena does not wants u-boot 's output on normal UART.

perezmeyer replied:
But I'll not apply it until the rest works

delhiryder commented:
Thanks @perezmeyer for your inputs. One of the biggest  changes I made was to drop the linux-beagleboard recipe in favor of the linux-bb.org recipe (this is what the stock meta-ti-bsp layer for the beagleplay board also uses). 

Is that a bad idea ? (especially since the .bb file has the all important `inherit kernel kernel-resin` line in it.

I had to do this initially because of all the build errors I was getting in moving from `u-boot` to `u-boot-ti-staging`

perezmeyer commented:
> Thanks @perezmeyer for your inputs. One of the biggest changes I made was to drop the linux-beagleboard recipe in favor of the linux-bb.org recipe (this is what the stock meta-ti-bsp layer for the beagleplay board also uses).
> 
> Is that a bad idea ? (especially since the .bb file has the all important `inherit kernel kernel-resin` line in it.
> 
> I had to do this initially because of all the build errors I was getting in moving from `u-boot` to `u-boot-ti-staging`

No, that's actually fine. I could boot the Beaglebone AI64 from an SD card using meta-ti's kernel and bootloader, this last one with some patches from the beaglebone one. They actually have almost no difference!

Now I could not boot from the eMMC, but looking at the differences it might explain why.

You could try creating a u-boot-ti-staging.bbappend file with at least:


```
# Use BeagleBone's u-boot until u-boot-ti-staging has the BBAI64 support upstreamed.
BRANCH:beaglebone-play= "v2023.04-ti-09.00.00.007"
UBOOT_GIT_URI:beaglebone-play = "git://git.beagleboard.org/beagleboard/u-boot.git"
SRCREV:beaglebone-play = "354f64f67e5b36df0027dbfbf071503b0071f4ed"
```
And check what happens

delhiryder commented:
@perezmeyer - thanks for your prompt response ! I'll try this next week (travelling tomorrow).

delhiryder commented:
@perezmeyer I (finally) got the balena OS boot sequence kicked off. Unfortunately, it fails to run to completion (see logs below):

```
Starting kernel ...

I/TC: Secondary CPU 1 initializing
I/TC: Secondary CPU 1 switching to normal world boot
I/TC: Secondary CPU 2 initializing
I/TC: Secondary CPU 2 switching to normal world boot
I/TC: Secondary CPU 3 initializing
I/TC: Secondary CPU 3 switching to normal world boot
[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x410fd034]
[    0.000000] Linux version 5.10.162+ (oe-user@oe-host) (aarch64-poky-linux-gcc (GCC) 11.3.0, GNU ld (GNU Binutils) 2.38.20220708) #1 SMP Thu Apr 6 21:42:04 UTC 2023
[    0.000000] Machine model: BeagleBoard.org BeaglePlay
[    0.000000] efi: UEFI not found.
[    0.000000] Reserved memory: created DMA memory pool at 0x000000009c800000, size 3 MiB
[    0.000000] OF: reserved mem: initialized node ipc-memories@9c800000, compatible id shared-dma-pool
[    0.000000] Reserved memory: created DMA memory pool at 0x000000009cb00000, size 1 MiB
[    0.000000] OF: reserved mem: initialized node m4f-dma-memory@9cb00000, compatible id shared-dma-pool
[    0.000000] Reserved memory: created DMA memory pool at 0x000000009cc00000, size 14 MiB
[    0.000000] OF: reserved mem: initialized node m4f-memory@9cc00000, compatible id shared-dma-pool
[    0.000000] Reserved memory: created DMA memory pool at 0x000000009da00000, size 1 MiB
[    0.000000] OF: reserved mem: initialized node r5f-dma-memory@9da00000, compatible id shared-dma-pool
[    0.000000] Reserved memory: created DMA memory pool at 0x000000009db00000, size 12 MiB
[    0.000000] OF: reserved mem: initialized node r5f-memory@9db00000, compatible id shared-dma-pool
[    0.000000] Zone ranges:
[    0.000000]   DMA      [mem 0x0000000080000000-0x00000000ffffffff]
[    0.000000]   DMA32    empty
[    0.000000]   Normal   empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000080000000-0x000000009c7fffff]
[    0.000000]   node   0: [mem 0x000000009c800000-0x000000009e6fffff]
[    0.000000]   node   0: [mem 0x000000009e700000-0x000000009e77ffff]
[    0.000000]   node   0: [mem 0x000000009e780000-0x000000009fffffff]
[    0.000000]   node   0: [mem 0x00000000a0000000-0x00000000ffffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000080000000-0x00000000ffffffff]
[    0.000000] cma: Reserved 144 MiB at 0x00000000f4000000
[    0.000000] psci: probing for conduit method from DT.
[    0.000000] psci: PSCIv1.1 detected in firmware.
[    0.000000] psci: Using standard PSCI v0.2 function IDs
[    0.000000] psci: Trusted OS migration not required
[    0.000000] psci: SMC Calling Convention v1.2
[    0.000000] percpu: Embedded 32 pages/cpu s92952 r8192 d29928 u131072
[    0.000000] Detected VIPT I-cache on CPU0
[    0.000000] CPU features: detected: ARM erratum 845719
[    0.000000] CPU features: detected: GIC system register CPU interface
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 516096
[    0.000000] Kernel command line: console=ttyS2,115200n8 root=PARTUUID=2f8cd71b-02 rw rootfstype=ext4 rootwait
[    0.000000] Dentry cache hash table entries: 262144 (order: 9, 2097152 bytes, linear)
[    0.000000] Inode-cache hash table entries: 131072 (order: 8, 1048576 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:on, heap free:off
[    0.000000] Memory: 1764708K/2097152K available (16704K kernel code, 1996K rwdata, 5132K rodata, 62272K init, 693K bss, 184988K reserved, 147456K cma-reserved)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000] rcu:     RCU event tracing is enabled.
[    0.000000] rcu:     RCU restricting CPUs from NR_CPUS=256 to nr_cpu_ids=4.
[    0.000000]  Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 25 jiffies.
[    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=4
[    0.000000] NR_IRQS: 64, nr_irqs: 64, preallocated irqs: 0
[    0.000000] GICv3: GIC: Using split EOI/Deactivate mode
[    0.000000] GICv3: 256 SPIs implemented
[    0.000000] GICv3: 0 Extended SPIs implemented
[    0.000000] GICv3: Distributor has no Range Selector support
[    0.000000] GICv3: 16 PPIs implemented
[    0.000000] GICv3: CPU0: found redistributor 0 region 0:0x0000000001880000
[    0.000000] ITS [mem 0x01820000-0x0182ffff]
[    0.000000] GIC: enabling workaround for ITS: Socionext Synquacer pre-ITS
[    0.000000] ITS@0x0000000001820000: Devices Table too large, reduce ids 20->19
[    0.000000] ITS@0x0000000001820000: allocated 524288 Devices @80800000 (flat, esz 8, psz 64K, shr 0)
[    0.000000] ITS: using cache flushing for cmd queue
[    0.000000] GICv3: using LPI property table @0x00000000800c0000
[    0.000000] GIC: using cache flushing for LPI property table
[    0.000000] GICv3: CPU0: using allocated LPI pending table @0x00000000800d0000
[    0.000000] arch_timer: cp15 timer(s) running at 200.00MHz (phys).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0x2e2049d3e8, max_idle_ns: 440795210634 ns
[    0.000004] sched_clock: 56 bits at 200MHz, resolution 5ns, wraps every 4398046511102ns
[    0.000361] Console: colour dummy device 80x25
[    0.000421] Calibrating delay loop (skipped), value calculated using timer frequency.. 400.00 BogoMIPS (lpj=800000)
[    0.000434] pid_max: default: 32768 minimum: 301
[    0.000721] LSM: Security Framework initializing
[    0.000771] Yama: becoming mindful.
[    0.000919] AppArmor: AppArmor initialized
[    0.000928] TOMOYO Linux initialized
[    0.001056] Mount-cache hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.001079] Mountpoint-cache hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.003490] rcu: Hierarchical SRCU implementation.
[    0.004814] Platform MSI: msi-controller@1820000 domain created
[    0.005111] PCI/MSI: /bus@f0000/interrupt-controller@1800000/msi-controller@1820000 domain created
[    0.005180] EFI services will not be available.
[    0.005864] smp: Bringing up secondary CPUs ...
[    0.014644] Detected VIPT I-cache on CPU1
[    0.014683] GICv3: CPU1: found redistributor 1 region 0:0x00000000018a0000
[    0.014697] GICv3: CPU1: using allocated LPI pending table @0x00000000800e0000
[    0.014765] CPU1: Booted secondary processor 0x0000000001 [0x410fd034]
[    0.023611] Detected VIPT I-cache on CPU2
[    0.023638] GICv3: CPU2: found redistributor 2 region 0:0x00000000018c0000
[    0.023649] GICv3: CPU2: using allocated LPI pending table @0x00000000800f0000
[    0.023689] CPU2: Booted secondary processor 0x0000000002 [0x410fd034]
[    0.032532] Detected VIPT I-cache on CPU3
[    0.032559] GICv3: CPU3: found redistributor 3 region 0:0x00000000018e0000
[    0.032570] GICv3: CPU3: using allocated LPI pending table @0x0000000080100000
[    0.032608] CPU3: Booted secondary processor 0x0000000003 [0x410fd034]
[    0.032703] smp: Brought up 1 node, 4 CPUs
[    0.032739] SMP: Total of 4 processors activated.
[    0.032746] CPU features: detected: 32-bit EL0 Support
[    0.032753] CPU features: detected: CRC32 instructions
[    0.032759] CPU features: detected: 32-bit EL1 Support
[    0.048201] CPU: All CPU(s) started at EL2
[    0.048270] alternatives: patching kernel code
[    0.049986] devtmpfs: initialized
[    0.060222] KASLR disabled due to lack of seed
[    0.060531] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.060580] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.065237] pinctrl core: initialized pinctrl subsystem
[    0.065941] DMI not present or invalid.
[    0.066730] NET: Registered protocol family 16
[    0.068699] DMA: preallocated 256 KiB GFP_KERNEL pool for atomic allocations
[    0.068862] DMA: preallocated 256 KiB GFP_KERNEL|GFP_DMA pool for atomic allocations
[    0.069041] DMA: preallocated 256 KiB GFP_KERNEL|GFP_DMA32 pool for atomic allocations
[    0.069121] audit: initializing netlink subsys (disabled)
[    0.069407] audit: type=2000 audit(0.068:1): state=initialized audit_enabled=0 res=1
[    0.069989] thermal_sys: Registered thermal governor 'fair_share'
[    0.069993] thermal_sys: Registered thermal governor 'bang_bang'
[    0.069999] thermal_sys: Registered thermal governor 'step_wise'
[    0.070004] thermal_sys: Registered thermal governor 'user_space'
[    0.070008] thermal_sys: Registered thermal governor 'power_allocator'
[    0.070697] hw-breakpoint: found 6 breakpoint and 4 watchpoint registers.
[    0.070839] ASID allocator initialised with 65536 entries
[    0.099995] HugeTLB registered 1.00 GiB page size, pre-allocated 0 pages
[    0.100017] HugeTLB registered 32.0 MiB page size, pre-allocated 0 pages
[    0.100023] HugeTLB registered 2.00 MiB page size, pre-allocated 0 pages
[    0.100029] HugeTLB registered 64.0 KiB page size, pre-allocated 0 pages
[    0.101885] cryptd: max_cpu_qlen set to 1000
[    0.105370] raid6: skip pq benchmark and using algorithm neonx8
[    0.105392] raid6: using neon recovery algorithm
[    0.106329] k3-chipinfo 43000014.chipid: Family:AM62X rev:SR1.0 JTAGID[0x0bb7e02f] Detected
[    0.106807] vdd_3v3: supplied by vsys_5v0
[    0.107759] iommu: Default domain type: Translated
[    0.108563] SCSI subsystem initialized
[    0.108781] usbcore: registered new interface driver usbfs
[    0.108828] usbcore: registered new interface driver hub
[    0.108863] usbcore: registered new device driver usb
[    0.109390] mc: Linux media interface: v0.10
[    0.109416] videodev: Linux video capture interface: v2.00
[    0.109544] pps_core: LinuxPPS API ver. 1 registered
[    0.109550] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.109565] PTP clock support registered
[    0.109600] EDAC MC: Ver: 3.0.0
[    0.110387] omap-mailbox 29000000.mailbox: omap mailbox rev 0x66fc9100
[    0.111589] NetLabel: Initializing
[    0.111601] NetLabel:  domain hash size = 128
[    0.111605] NetLabel:  protocols = UNLABELED CIPSOv4 CALIPSO
[    0.111672] NetLabel:  unlabeled traffic allowed by default
[    0.112426] clocksource: Switched to clocksource arch_sys_counter
[    0.219359] VFS: Disk quotas dquot_6.6.0
[    0.219453] VFS: Dquot-cache hash table entries: 512 (order 0, 4096 bytes)
[    0.219645] FS-Cache: Loaded
[    0.220085] CacheFiles: Loaded
[    0.220660] AppArmor: AppArmor Filesystem Enabled
[    0.227805] NET: Registered protocol family 2
[    0.228076] IP idents hash table entries: 32768 (order: 6, 262144 bytes, linear)
[    0.229433] tcp_listen_portaddr_hash hash table entries: 1024 (order: 2, 16384 bytes, linear)
[    0.229666] TCP established hash table entries: 16384 (order: 5, 131072 bytes, linear)
[    0.229897] TCP bind hash table entries: 16384 (order: 6, 262144 bytes, linear)
[    0.230122] TCP: Hash tables configured (established 16384 bind 16384)
[    0.230476] MPTCP token hash table entries: 2048 (order: 3, 49152 bytes, linear)
[    0.230606] UDP hash table entries: 1024 (order: 3, 32768 bytes, linear)
[    0.230666] UDP-Lite hash table entries: 1024 (order: 3, 32768 bytes, linear)
[    0.230911] NET: Registered protocol family 1
[    0.230937] NET: Registered protocol family 44
[    0.230953] PCI: CLS 0 bytes, default 64
[    0.412730] hw perfevents: enabled with armv8_cortex_a53 PMU driver, 7 counters available
[    0.413346] kvm [1]: IPA Size Limit: 40 bits
[    0.414173] kvm [1]: vgic-v2@100010000
[    0.414201] kvm [1]: GIC system register CPU interface enabled
[    0.414321] kvm [1]: vgic interrupt IRQ9
[    0.414456] kvm [1]: Hyp mode initialized successfully
[    0.419185] Initialise system trusted keyrings
[    0.419429] workingset: timestamp_bits=46 max_order=19 bucket_order=0
[    0.425020] zbud: loaded
[    0.426735] jffs2: version 2.2. (NAND) (SUMMARY)  © 2001-2006 Red Hat, Inc.
[    0.427899] fuse: init (API version 7.32)
[    0.464441] xor: measuring software checksum speed
[    0.468482]    8regs           :  2464 MB/sec
[    0.471857]    32regs          :  2919 MB/sec
[    0.475761]    arm64_neon      :  2527 MB/sec
[    0.475766] xor: using function: 32regs (2919 MB/sec)
[    0.475784] Key type asymmetric registered
[    0.475791] Asymmetric key parser 'x509' registered
[    0.475849] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 243)
[    0.476059] io scheduler mq-deadline registered
[    0.478255] pinctrl-single 4084000.pinctrl: 34 pins, size 136
[    0.478775] pinctrl-single f4000.pinctrl: 171 pins, size 684
[    0.487187] Serial: 8250/16550 driver, 10 ports, IRQ sharing enabled
[    0.501945] brd: module loaded
[    0.640534] loop: module loaded
[    0.641132] zram: Added device: zram0
[    0.642466] megasas: 07.714.04.00-rc1
[    0.647255] tun: Universal TUN/TAP device driver, 1.6
[    0.647890] igbvf: Intel(R) Gigabit Virtual Function Network Driver
[    0.647899] igbvf: Copyright (c) 2009 - 2012 Intel Corporation.
[    0.647996] sky2: driver version 1.30
[    0.649062] VFIO - User Level meta-driver version: 0.3
[    0.649659] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    0.649738] ehci-pci: EHCI PCI platform driver
[    0.650339] usbcore: registered new interface driver uas
[    0.650380] usbcore: registered new interface driver usb-storage
[    0.650474] usbcore: registered new interface driver usbserial_generic
[    0.650518] usbserial: USB Serial support registered for generic
[    0.652142] i2c /dev entries driver
[    0.652365] Driver for 1-wire Dallas network protocol.
[    0.652491] DS1WM w1 busmaster driver - (c) 2004 Szabolcs Gyurko
[    0.654405] sdhci: Secure Digital Host Controller Interface driver
[    0.654419] sdhci: Copyright(c) Pierre Ossman
[    0.654719] sdhci-pltfm: SDHCI platform and OF driver helper
[    0.655848] ledtrig-cpu: registered to indicate activity on CPUs
[    0.656284] SMCCC: SOC_ID: ARCH_SOC_ID not implemented, skipping ....
[    0.656821] hid: raw HID events driver (C) Jiri Kosina
[    0.657071] usbcore: registered new interface driver usbhid
[    0.657077] usbhid: USB HID core driver
[    0.657921] optee: probing for conduit method.
[    0.657960] optee: revision 3.18 (4adb4a1f)
[    0.658364] optee: dynamic shared memory is enabled
[    0.658598] optee: initialized driver
[    0.659554] drop_monitor: Initializing network drop monitor service
[    0.660183] IPVS: Registered protocols (TCP, UDP, SCTP, AH, ESP)
[    0.660245] IPVS: Connection hash table configured (size=4096, memory=32Kbytes)
[    0.660472] IPVS: ipvs loaded.
[    0.661254] NET: Registered protocol family 10
[    0.681731] Segment Routing with IPv6
[    0.681849] mip6: Mobile IPv6
[    0.681963] NET: Registered protocol family 17
[    0.682052] bridge: filtering via arp/ip/ip6tables is no longer available by default. Update your scripts to load br_netfilter if you need this.
[    0.682240] Key type dns_resolver registered
[    0.682248] mpls_gso: MPLS GSO support
[    0.682586] registered taskstats version 1
[    0.682603] Loading compiled-in X.509 certificates
[    0.682780] zswap: loaded using pool lzo/zbud
[    0.683197] Key type .fscrypt registered
[    0.683206] Key type fscrypt-provisioning registered
[    0.688306] Btrfs loaded, crc32c=crc32c-generic
[    0.688504] AppArmor: AppArmor sha1 policy hashing enabled
[    0.699831] ti-sci 44043000.system-controller: ABI: 3.1 (firmware rev 0x0008 '8.5.3--v08.05.03 (Chill Capybar')
[    0.767369] omap-gpmc 3b000000.memory-controller: GPMC revision 6.0
[    0.767394] gpmc_mem_init: disabling cs 0 mapped at 0x0-0x1000000
[    0.770770] omap_i2c 4900000.i2c: bus 5 rev0.12 at 10 kHz
[    0.772780] at24 0-0050: supply vcc not found, using dummy regulator
[    0.796611] at24 0-0050: 4096 byte 24c32 EEPROM, writable, 1 bytes/write
[    0.800222] VDD_CORE: Bringing 750000uV into 850000-850000uV
[    0.801092] VDD_CORE: supplied by vsys_5v0
[    0.802043] VDD_1V8: supplied by vsys_5v0
[    0.802932] VDD_1V2: supplied by vsys_5v0
[    0.803967] VDD_SD_3V3: supplied by vdd_3v3
[    0.805171] VDDA_0V85: supplied by VDD_1V8
[    0.806342] VDDA_1V8: supplied by vdd_3v3
[    0.807218] VDD_2V5: supplied by vdd_3v3
[    0.809737] input: tps65219-pwrbutton as /devices/platform/bus@f0000/20000000.i2c/i2c-0/0-0030/tps65219-pwrbutton.1.auto/input/input0
[    0.810475] tps65219 0-0030: Setup as system-power-controller but pm_power_off function already registered, overwriting
[    0.810752] omap_i2c 20000000.i2c: bus 0 rev0.12 at 400 kHz
[    0.812443] omap_i2c 20010000.i2c: bus 1 rev0.12 at 10 kHz
[    0.814405] omap_i2c 20020000.i2c: bus 2 rev0.12 at 10 kHz
[    0.815804] omap_i2c 20030000.i2c: bus 3 rev0.12 at 400 kHz
[    0.816213] ti-sci-intr 4210000.interrupt-controller: Interrupt Router 5 domain created
[    0.816383] ti-sci-intr bus@f0000:interrupt-controller@a00000: Interrupt Router 3 domain created
[    0.816691] ti-sci-inta 48000000.interrupt-controller: Interrupt Aggregator domain 28 created
[    0.817018] ti-udma 485c0100.dma-controller: Number of rings: 82
[    0.819274] ti-udma 485c0100.dma-controller: Channels: 48 (bchan: 18, tchan: 12, rchan: 18)
[    0.822568] ti-udma 485c0000.dma-controller: Number of rings: 150
[    0.826994] ti-udma 485c0000.dma-controller: Channels: 35 (tchan: 20, rchan: 15)
[    0.830431] 4a00000.serial: ttyS1 at MMIO 0x4a00000 (irq = 15, base_baud = 3000000) is a 8250
[    0.831944] 2b300000.serial: ttyS0 at MMIO 0x2b300000 (irq = 18, base_baud = 3000000) is a 8250
[    0.833400] 2800000.serial: ttyS2 at MMIO 0x2800000 (irq = 29, base_baud = 3000000) is a 8250
[    2.358942] printk: console [ttyS2] enabled
[    2.365007] 2850000.serial: ttyS3 at MMIO 0x2850000 (irq = 30, base_baud = 3000000) is a 8250
[    2.373614] serial serial0: tty port ttyS3 registered
[    2.379430] omap8250 2860000.serial: PM domain pd:158 will not be powered off
[    2.386905] 2860000.serial: ttyS4 at MMIO 0x2860000 (irq = 31, base_baud = 3000000) is a 8250
[    2.395575] serial serial1: tty port ttyS4 registered
[    2.406575] am65-cpsw-nuss 8000000.ethernet: initializing am65 cpsw nuss version 0x6BA01103, cpsw version 0x6BA81103 Ports: 3 quirks:00000006
[    2.419584] am65-cpsw-nuss 8000000.ethernet: Use random MAC address
[    2.425854] am65-cpsw-nuss 8000000.ethernet: initialized cpsw ale version 1.5
[    2.432978] am65-cpsw-nuss 8000000.ethernet: ALE Table size 512
[    2.439742] am65-cpsw-nuss 8000000.ethernet: CPTS ver 0x4e8a010c, freq:500000000, add_val:1 pps:0
[    2.450268] am65-cpsw-nuss 8000000.ethernet: set new flow-id-base 19
[    2.461445] rtc-ti-k3 2b1f0000.rtc: registered as rtc1
[    2.570494] sdhci-am654 fa00000.mmc: parsing dt failed (-517)
[    2.570913] mmc0: CQHCI version 5.10
[    2.580505] wlan_en: supplied by vdd_3v3
[    2.585022] vdd_3v3_sd: supplied by vdd_3v3
[    2.590248] mikrobus:mikrobus_port_register: registering port mikrobus-0
[    2.590315] mikrobus mikrobus-0: mikrobus port 0 eeprom empty probing default eeprom
[    2.606367] bq32k 0-0068: Oscillator Failure. Check RTC battery.
[    2.612899] bq32k 0-0068: registered as rtc0
[    2.616469] mmc0: SDHCI controller on fa10000.mmc [fa10000.mmc] using ADMA 64-bit
[    2.617443] bq32k 0-0068: hctosys: unable to read the hardware clock
[    2.694585] mmc0: Command Queue Engine enabled
[    2.699059] mmc0: new HS200 MMC card at address 0001
[    2.704931] mmcblk0: mmc0:0001 TB2916 14.6 GiB
[    2.709733] mmcblk0boot0: mmc0:0001 TB2916 partition 1 4.00 MiB
[    2.715939] mmcblk0boot1: mmc0:0001 TB2916 partition 2 4.00 MiB
[    2.717451] it66121 2-004c: IT66121 revision 1 probed
[    2.722073] mmcblk0rpmb: mmc0:0001 TB2916 partition 3 4.00 MiB, chardev (236:0)
[    2.735876]  mmcblk0: p1 p2
[    2.952300] [drm] Initialized tidss 1.0.0 20180215 for 30200000.dss on minor 0
[    2.967854] tidss 30200000.dss: [drm] Cannot find any crtc or sizes
[    3.123065] debugfs: Directory 'pd:53' with parent 'pm_genpd' already present!
[    3.123423] sdhci-am654 fa00000.mmc: Got CD GPIO
[    3.123851] mmc2: CQHCI version 5.10
[    3.130416] debugfs: Directory 'pd:52' with parent 'pm_genpd' already present!
[    3.135477] mmc1: CQHCI version 5.10
[    3.138548] debugfs: Directory 'pd:51' with parent 'pm_genpd' already present!
[    3.156914] debugfs: Directory 'pd:182' with parent 'pm_genpd' already present!
[    3.165310] mmc2: SDHCI controller on fa20000.mmc [fa20000.mmc] using ADMA 64-bit
[    3.167141] input: gpio-keys as /devices/platform/gpio-keys/input/input1
[    3.178655] sdhci-am654 fa20000.mmc: card claims to support voltages below defined range
[    3.179984] of_cfs_init
[    3.184765] mmc1: SDHCI controller on fa00000.mmc [fa00000.mmc] using ADMA 64-bit
[    3.197662] of_cfs_init: OK
[    3.199594] mmc2: new high speed SDIO card at address 0001
[    3.265535] Freeing unused kernel memory: 62272K
[    3.295107] Checked W+X mappings: passed, no W+X pages found
[    3.298830] mmc1: new ultra high speed SDR104 SDHC card at address 0001
[    3.300803] Run /init as init process
[    3.308774] mmcblk1: mmc1:0001 SD32G 28.8 GiB
[    3.319949]  mmcblk1: p1 p2 p3 p4 < p5 p6 >
Starting version 250.5+
[    4.129872] zram0: detected capacity change from 0 to 1010913280
[    4.199021] w1_master_driver w1_bus_master1: w1_search: max_slave_count 64 reached, will continue next search.
[   10.954740] EXT4-fs (mmcblk1p2): mounted filesystem with ordered data mode. Opts: (null)
[   11.031350] EXT4-fs (mmcblk1p5): mounted filesystem with ordered data mode. Opts: (null)
[   11.397149] Not activating Mandatory Access Control as /sbin/tomoyo-init does not exist.
[   13.280435] random: crng init done
[   13.293502] EXT4-fs (mmcblk1p2): re-mounted. Opts: (null)
[   13.364804] Not activating Mandatory Access Control as /sbin/tomoyo-init does not exist.
[   13.555653] systemd[1]: System time before build time, advancing clock.

Welcome to balenaOS 3.1.8!

[   13.741436] systemd[686]: /lib/systemd/system-generators/systemd-gpt-auto-generator failed with exit status 1.
[   14.026876] systemd[1]: /lib/systemd/system/balena-host.socket:5: ListenStream= references a path below legacy directory /var/run/, updating /var/run/balena-host.sock → /run/balena-host.sock; please update the unit file accordingly.
[   14.081965] systemd[1]: /lib/systemd/system/balena-engine.socket:7: ListenStream= references a path below legacy directory /var/run/, updating /var/run/balena-engine.sock → /run/balena-engine.sock; please update the unit file accordingly.
[  OK  ] Created slice Slice /system/balena-info.
[  OK  ] Created slice Slice /system/getty.
[  OK  ] Created slice Slice /system/modprobe.
[  OK  ] Created slice Slice /system/serial-getty.
[  OK  ] Created slice User and Session Slice.
[  OK  ] Started Forward Password R…uests to Wall Directory Watch.
[  OK  ] Reached target Development features.
[  OK  ] Reached target Remote File Systems.
[  OK  ] Reached target Slice Units.
[  OK  ] Reached target Swaps.
[  OK  ] Listening on initctl Compatibility Named Pipe.
[  OK  ] Listening on Journal Socket (/dev/log).
[  OK  ] Listening on Journal Socket.
[  OK  ] Listening on udev Control Socket.
[  OK  ] Listening on udev Kernel Socket.
[  OK  ] Listening on User Database Manager Socket.
         Mounting Huge Pages File System...
         Mounting POSIX Message Queue File System...
         Mounting Kernel Debug File System...
         Mounting Kernel Trace File System...
         Mounting Temporary Directory /tmp...
         Starting Create List of Static Device Nodes...
         Starting Resin active root partition mount service...
         Starting Load Kernel Module configfs...
         Starting Load Kernel Module drm...
         Starting Load Kernel Module fuse...
         Starting Resin boot partition mount service...
         Starting Resin state partition mount service...
         Starting Rollback clear bo…boot crumb is still present...
         Starting Generate network …ts from Kernel command line...
         Starting Remount Root and Kernel File Systems...
         Starting Apply Kernel Variables...
         Starting Coldplug All udev Devices...
         Starting Setup Virtual Console...
         Starting Set the system clock from the build timestamp...
[  OK  ] Mounted Huge Pages File System.
[  OK  ] Mounted POSIX Message Queue File System.
[  OK  ] Mounted Kernel Debug File System.
[  OK  ] Mounted Kernel Trace File System.
[  OK  ] Mounted Temporary Directory /tmp.
[  OK  ] Finished Create List of Static Device Nodes.
[  OK  ] Finished Load Kernel Module configfs.
[  OK  ] Finished Load Kernel Module drm.
[  OK  ] Finished Load Kernel Module fuse.
[  OK  ] Finished Resin boot partition mount service.
[  OK  ] Finished Resin state partition mount service.
[  OK  ] Finished Rollback clear bo…ltboot crumb is still present.
[  OK  ] Finished Generate network units from Kernel command line.
[  OK  ] Finished Remount Root and Kernel File Systems.
[  OK  ] Finished Apply Kernel Variables.
[  OK  ] Finished Setup Virtual Console.
[  OK  ] Finished Set the system clock from the build timestamp.
         Mounting FUSE Control File System...
         Mounting Kernel Configuration File System...
         Starting OS configuration device key service...
         Starting Expand resin-data partition...
         Starting Create Static Device Nodes in /dev...
[  OK  ] Mounted FUSE Control File System.
[  OK  ] Mounted Kernel Configuration File System.
[  OK  ] Finished Coldplug All udev Devices.
[  OK  ] Finished Create Static Device Nodes in /dev.
[  OK  ] Finished OS configuration device key service.
[  OK  ] Reached target Preparation for Local File Systems.
[  OK  ] Set up automount mnt-sysroot-inactive partition.
         Mounting /var/volatile...
         Starting Wait for udev To …plete Device Initialization...
[  OK  ] Finished Expand resin-data partition.
[  OK  ] Mounted /var/volatile.
         Starting Runtime development features management...
         Starting Resin data partition mount service...
         Mounting Bind mount for /etc/fake-hwclock...
         Starting Bind mount volatile /var/cache...
         Starting Bind mount volatile /var/lib...
         Starting Bind mount volatile /var/spool...
         Starting Bind mount volatile /srv...
[  OK  ] Finished Resin data partition mount service.
[  OK  ] Mounted Bind mount for /etc/fake-hwclock.
[  OK  ] Finished Bind mount volatile /var/cache.
[  OK  ] Finished Bind mount volatile /var/lib.
[  OK  ] Finished Bind mount volatile /var/spool.
[  OK  ] Finished Bind mount volatile /srv.
[  OK  ] Finished Runtime development features management.
         Mounting resin-data bind mount...
         Mounting Docker data mountpoint...
         Starting Restore/save the current clock...
[  OK  ] Mounted resin-data bind mount.
[  OK  ] Mounted Docker data mountpoint.
[  OK  ] Finished Restore/save the current clock.
[  OK  ] Reached target Local File Systems.
         Starting Tell Plymouth To Write Out Runtime Data...
         Starting Journal Service...
[  OK  ] Finished Tell Plymouth To Write Out Runtime Data.
[  OK  ] Started Journal Service.
         Starting Create Volatile Files and Directories...
[  OK  ] Finished Create Volatile Files and Directories.
         Starting Bind mount for /etc/NetworkManager/conf.d...
         Starting Bind mount for /e…NetworkManager/dispatcher.d...
         Starting Bind mount for /e…kManager/system-connections...
         Starting Bind mount for /etc/balena-supervisor...
         Starting Bind mount for /etc/docker...
         Starting Bind mount for /etc/hostname...
         Starting Bind mount for /etc/openvpn...
         Starting Bind mount for /etc/ssh/hostkeys...
         Starting Bind mount for /etc/udev/rules.d...
         Starting Bind mount for /home/root/.docker...
         Starting Bind mount for /home/root/.rnd...
         Starting Bind mount for /home/root/.ssh...
         Starting Bind mount for /u…hare/ca-certificates/balena...
         Starting Bind mount for /var/lib/NetworkManager...
         Starting Bind mount for /var/lib/bluetooth...
         Starting Bind mount for /var/lib/chrony...
         Starting Bind mount for /var/lib/systemd...
         Starting Record System Boot/Shutdown in UTMP...
[  OK  ] Finished Bind mount for /etc/NetworkManager/conf.d.
[  OK  ] Finished Bind mount for /etc/NetworkManager/dispatcher.d.
[  OK  ] Finished Bind mount for /e…orkManager/system-connections.
[  OK  ] Finished Bind mount for /etc/balena-supervisor.
[  OK  ] Finished Bind mount for /etc/docker.
[  OK  ] Finished Bind mount for /etc/hostname.
[  OK  ] Finished Bind mount for /etc/openvpn.
[  OK  ] Finished Bind mount for /etc/ssh/hostkeys.
[  OK  ] Finished Bind mount for /etc/udev/rules.d.
[  OK  ] Finished Bind mount for /home/root/.docker.
[  OK  ] Finished Bind mount for /home/root/.rnd.
[  OK  ] Finished Bind mount for /home/root/.ssh.
[  OK  ] Finished Bind mount for /u…/share/ca-certificates/balena.
[  OK  ] Finished Bind mount for /var/lib/NetworkManager.
[  OK  ] Finished Bind mount for /var/lib/bluetooth.
[  OK  ] Finished Bind mount for /var/lib/chrony.
[  OK  ] Finished Bind mount for /var/lib/systemd.
         Starting Load/Save Random Seed...
         Starting Rule-based Manage…for Device Events and Files...
[  OK  ] Finished Load/Save Random Seed.
[  OK  ] Finished Record System Boot/Shutdown in UTMP.
[  OK  ] Started Rule-based Manager for Device Events and Files.
         Starting Show Plymouth Boot Screen...
[   19.274055] k3-m4-rproc 5000000.m4fss: configured M4 for remoteproc mode
[   19.301257] remoteproc remoteproc0: request_firmware failed: -2
[   19.327540] platform 78000000.r5f: could not request atcm region for resource
[   19.334850] platform 78000000.r5f: failed to get internal memories, ret = -16
[   19.342181] k3_r5_rproc bus@f0000:bus@b00000:r5fss@78000000: k3_r5_core_of_init failed, ret = -16
[   19.351226] k3_r5_rproc bus@f0000:bus@b00000:r5fss@78000000: k3_r5_cluster_of_init failed, ret = -16
[   19.424280] davinci-mcasp 2b10000.mcasp: IRQ common not found
[   20.389047] hub 2-0:1.0: config failed, hub doesn't have any ports! (err -19)
[   41.334066] wl1271_sdio mmc2:0001:2: sdio write failed (-110)
```

After the last error message, the system reboots.

Any ideas what might be going on here ?

delhiryder commented:
Thanks @perezmeyer - which discord server/channel should I try ?

(btw, I applied your bb-append change to get the u-boot-ti-staging sources from the beagleboard.org gitlab repo, but the failure mode is the same)

perezmeyer commented:
@delhiryder beagleboard.org in there you will see a beagleplay channel

perezmeyer commented on layers/meta-balena-beaglebone/recipes-kernel/linux/linux-bb.org_git.bbappend:
```diff
@@ -1 +1,27 @@
+SRCREV:aarch64 = "51c09b7d81b4531f4846cb6c24f46ac239acd69d"
+PV:aarch64 = "6.1.46+git${SRCPV}"
+
+KERNEL_DEVICETREE:aarch64 = " \
```
aarch64 might probably be too broad here.

delhiryder replied:
Thanks @perezmeyer : Any suggestions on how to narrow this down ? (can I actually use the `beaglebone-play*` targets ?

delhiryder commented:
@perezmeyer Quick update: After applying patches to the `linux-bb.org`, `u-boot-ti-staging` **and `linux-firmware` recipes, based upon suggestions from the good folks at the `BeaglePlay` channel on the `Beagleboard` discord server, I was *still* seeing the same errors and failures. Thankfully, someone thought that it may be a brownout/undervoltage issue, and suggested a power supply with more wattage. I ordered an official Raspberry Pi4 power supply, and the reboot problem has gone away. 

Going to be pushing some more changes here (related to updated recipes and the patch to the `DISTRO_BOOTCMD`.

perezmeyer commented:
@delhiryder You wrote:

> [90f108b](https://github.com/balena-os/balena-beaglebone/commit/90f108b444ffbfda6a0bd303c4a29f512919c255) Checkpoint: Changed the [linux-bb.org](http://linux-bb.org/), linux-firmware and u-boot-ti-staging tags (to try and get rid of some errors). Yocto build is failing, but should be (relatively) easy to fix.

Is it perhaps failing like in https://github.com/balena-os/balena-beaglebone/pull/586#issuecomment-1747074441 ?

delhiryder commented:
@perezmeyer I think I got past the latest (yocto) build error. I am now trying to get the following warning/errors resolved:

```
[  367.720466] wl1271_sdio mmc2:0001:2: sdio write failed (-110)
[  376.982472] wl1271_sdio mmc2:0001:2: sdio write failed (-110)
[  386.232489] wl1271_sdio mmc2:0001:2: sdio write failed (-110)
[  386.897179] wlcore: ERROR firmware boot failed despite 3 retries
```

These appear to *settle down* after about 400 seconds.

perezmeyer commented:
Excellent! Remember I am not part of the Balena stuff, but ping me if I can be of any help.

perezmeyer commented:
Be sure to rebase master in your branch. The beagleBone ai64 stuff is (almost all) there.
